packages/taco/src/conditions/base/index.ts

```diff
@@ -1,37 +1,70 @@
 import {
   CompoundConditionProps,
   compoundConditionSchema,
+  CompoundConditionType,
 } from '../compound-condition';
-import { Condition } from '../condition';
+import { Condition, ConditionProps } from '../condition';
 
-import { ContractConditionProps, contractConditionSchema } from './contract';
-import { RpcConditionProps, rpcConditionSchema } from './rpc';
-import { TimeConditionProps, timeConditionSchema } from './time';
+import { ContractConditionProps, contractConditionSchema, ContractConditionType } from './contract';
+import { RpcConditionProps, rpcConditionSchema, RpcConditionType } from './rpc';
+import { TimeConditionProps, timeConditionSchema, TimeConditionType } from './time';
+
+type OmitType<T> = Omit<T, 'conditionType'>
 
 // Exporting classes here instead of their respective schema files to
 // avoid circular dependency on Condition class.
 
 export class CompoundCondition extends Condition {
-  constructor(value: CompoundConditionProps) {
-    super(compoundConditionSchema, value);
+  constructor(value: OmitType<CompoundConditionProps>) {
+    super(compoundConditionSchema, {
+      conditionType: CompoundConditionType,
+      ...value,
+    });
+  }
+
+  private static withOperator(
+    operands: ConditionProps[],
+    operator: 'or' | 'and',
```

Comment:
Is there already an issue filed to deal with `not` compound conditions in 7.0.0? This is completed on the `nucypher` side -https://github.com/nucypher/nucypher/pull/3293

Reply:
Filed an issue here: https://github.com/nucypher/nucypher-ts/issues/369